proxy/server.py

```diff
@@ -291,12 +291,12 @@ def do_POST(self):
         contenttype = 'application/json'
         message = '{"error": "Invalid Request"}'
 
-        # Get the request path and subtract the base URL. This is intentionally doing a replace on the URL we check
-        # instead of using the prefix inside of the checks to maintain the ability to use the regular URLs in addition to
-        # the proxied URLs, so Telegraf configs or existing integrations don't need to be modified.
+        # If set, remove the api_base_url from the requested path. This allows installing the
+        # the proxy on a path without impacting the use of Telegraf or other integrations. Python 3.9+
         request_path = self.path
-        if request_path.startswith(api_base_url):
-            request_path = "/" + request_path[len(api_base_url):]
+        new_path = request_path.removeprefix(api_base_url)
+        if new_path is not request_path:
+            request_path = "/" + new_path
```

Comment:
I was a little concerned about the string manupation required for every GET and POST. I ran some tests to ensure we aren't adding too much. 

I ran different load tests using basic install with animation rendering. I measured with:

```bash
export PID=1234 # process ID of python server.py
top -b -p $PID -n 60 -d 1 | grep $PID | awk '{sum+=$9} END {print "Average CPU: " sum/NR "%"}'
```

* Before this PR
CPU Usage - Average: 3.145%

*  After PR
CPU Usage - Average: 4.875%

Since we are pinned to Python 3.10, we can use a more performant string editor for prefix.  I made that update and it brings us back down to about where we were.

* With this PR after optimization
CPU Usage - Average: 3.2%